src/federation_scheduled_tasks.rs

```diff
@@ -0,0 +1,106 @@
+use chrono::Duration;
+use clokwerk::{AsyncScheduler, TimeUnits};
+use diesel::{Connection, PgConnection, QueryDsl, RunQueryDsl};
+use lemmy_api_common::context::LemmyContext;
+use lemmy_apub::DEAD_INSTANCES;
+use lemmy_db_schema::{
+  schema::instance,
+  source::instance::{Instance, InstanceForm},
+  utils::naive_now,
+};
+use lemmy_routes::nodeinfo::NodeInfo;
+use lemmy_utils::error::{LemmyError, LemmyResult};
+use tokio::sync::OnceCell;
+use tracing::{error, info};
+
+pub async fn setup_federation_scheduled_tasks(
+  db_url: String,
+  context: LemmyContext,
+) -> Result<(), LemmyError> {
+  let mut scheduler = AsyncScheduler::new();
+
+  // Check for dead federated instances
+  static CONTEXT: OnceCell<LemmyContext> = OnceCell::const_new();
+  CONTEXT.set(context).ok();
+  static DB_URL: OnceCell<String> = OnceCell::const_new();
+  DB_URL.set(db_url).ok();
+  scheduler.every(TimeUnits::minutes(1)).run(|| async {
+    let mut conn =
+      PgConnection::establish(DB_URL.get().unwrap()).expect("could not establish connection");
+    // TODO: this is not getting executed for some reason. change to daily once working
+    check_dead_instances(&mut conn, CONTEXT.get().unwrap())
+      .await
+      .map_err(|e| error!("Failed to check federated instances: {e}"))
+      .ok();
+  });
+
+  // Mark instances which haven't been updated for three days or more as dead, and don't send
+  // any activities to them.
+  scheduler.every(TimeUnits::hour(1)).run(|| async {
+    let mut conn =
+      PgConnection::establish(DB_URL.get().unwrap()).expect("could not establish connection");
+    update_dead_instances(&mut conn).await;
+  });
+
+  // Manually run the scheduler in an event loop
+  tokio::spawn(async move {
```

Comment:
I'd make either both these scheduled things use tokio, or not. Not one use tokio and the other thread.